File: LeetCode/1351.Count-Negative-Numbers-in-a-Sorted-Matrix.cpp
// 1351. Count Negative Numbers in a Sorted Matrix

class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int ans = 0;
        int m = grid.size();
        int n = grid[0].size();
        for(int i=0; i<m; i++) {
            int low = 0;
            int high = n-1;
            while(low<=high) {
                int mid = (high-low)/2 + low;
                if(grid[i][mid]<0) {
                    high = mid-1;
                }
                else {
                    low = mid+1;
                }
            }
            ans += n-low;
        }
        return ans;
    }
};

File: LeetCode/1351-Count-Negative-Numbers-in-a-Sorted-Matrix.java
// 1351. Count Negative Numbers in a Sorted Matrix

class Solution {
    public int getInd(int[] arr, int n) {
        int ans = n;
        int l = 0, h = n - 1;
        while (l <= h) {
            int m = (h - l) / 2 + l;
            if (arr[m] < 0) {
                ans = m;
                h = m - 1;
            }
            else {
                l = m + 1;
            }
        }
        return ans;
    }

    public int countNegatives(int[][] grid) {
        int ans = 0, n = grid[0].length;
        for (int[] g : grid) {
            ans += n - getInd(g, n);
        }
        return ans;
    }
}
